File: services/post-service/src/main/java/com/socialmedia/postservice/dto/event/NotificationEvent.java
package com.socialmedia.postservice.dto.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class NotificationEvent {
    private String eventId;
    private String eventType;
    private String sourceService;
    private Instant timestamp;
    private String actorUserId;
    private String targetUserId;
    private String resourceType;
    private String resourceId;
    private Map<String, Object> metadata;
}

File: services/notification-service/Models/NotificationEvent.cs
namespace notification_service.Models;

public class NotificationEvent {
    public string EventId { get; set; } = null!;
    public string EventType { get; set; } = null!;
    public string SourceService { get; set; } = null!;
    public double Timestamp { get; set; }
    public string ActorUserId { get; set; } = null!;
    public string TargetUserId { get; set; } = null!;
    public string ResourceType { get; set; } = null!;
    public string ResourceId { get; set; } = null!;
    public Dictionary<string, object>? Metadata { get; set; }
}